envoy/stats/stats.h

```diff
@@ -209,5 +209,9 @@ using SizeFn = std::function<void(std::size_t)>;
  */
 template <typename Stat> using StatFn = std::function<void(Stat&)>;
 
+// Template that lazy-init a StatsStruct.
```

Comment:
s/lazy-init/lazily initializes/

Also add here that these are selected based on the bootstrap config setting.